parts/kubernetesmastervars.t

```diff
@@ -44,9 +44,8 @@
 {{end}}
 {{if .MasterProfile.IsCustomVNET}}
     "vnetSubnetID": "[parameters('masterVnetSubnetID')]",
-    "subnetName": "[parameters('masterVnetSubnetID')]",
-    "vnetParts": "[split(parameters('masterVnetSubnetID'),'/subnets/')]",
-    "virtualNetworkName": "[variables('vnetParts')[0]]",
+    "subnetName": "[split(parameters('masterVnetSubnetID'), '/')[10]]",
```

Comment:
I recommend replacing the numbers with a variable.  For example,  '10' with a variables('subnetNameResourceIndex') and '8' with a variables('virtualNetworkNameIndex').